src/index.js

```diff
@@ -1,14 +1,18 @@
 import behaviorSubject from 'callbag-behavior-subject'
-import { useRef } from 'react'
+import { useRef, useEffect } from 'react'
 
 export default function useValueCallbag(value) {
-  const value$Ref = useRef(null)
+  const value$Ref = useRef()
 
-  if (value$Ref.current === null) {
+  if (!value$Ref.current) {
     value$Ref.current = behaviorSubject(value)
-  } else {
-    value$Ref.current(1, value)
   }
 
+  useEffect(() => {
+    value$Ref.current(1, value)
```

Comment:
this will trigger an additional notification on "mount", is this OK for you? seems like a thing that you have wanted to avoid in the first place

Reply:
it seems like a single emission of the value of the initial value on mount ([playground](https://stackblitz.com/edit/react-g1nwpf?file=src%2FApp.js)), which is actually correct (otherwise you would not get any emissions for the initial value, as can be seen in the linked playground).

Reply:
The initial value is not ignored - this is the reason to use the behavior subject. It has a memory cell that tracks the last notification and "replays" it for all new subscribers:
https://stackblitz.com/edit/react-8fp5dm?file=src%2FApp.js

However, in this particular scenario that extra notification is just ignored, or rather, it's dispatched when no one is listening yet. This is for regular effects though - layout effects can still observe this little detail. Maybe it's not a big deal though? Having it here definitely simplifies the implementation.

Reply:
good catch! however [the fix](https://stackblitz.com/edit/react-rpstqx?file=src%2Fuse-value-callbag.js) does not seem to be particularly complicated, as this basically resolves the issue:

```ts
const lastVal = useRef(value)

useEffect(() => {
  if (value !== lastVal.current) {
    value$Ref.current(1, value)
    lastVal.current = value
  }
}, [value])
```

Reply:
but I also do agree that perhaps a redundant initial emission caught by layout effects is not really a big deal either.